clients/horizonclient/examples_test.go

```diff
@@ -1078,25 +1078,26 @@ func ExampleClient_SubmitTransaction() {
 		Asset:       txnbuild.NativeAsset{},
 	}
 
-	tx := txnbuild.Transaction{
-		SourceAccount: &sourceAccount,
-		Operations:    []txnbuild.Operation{&op},
-		Timebounds:    txnbuild.NewInfiniteTimeout(), // Use a real timeout in production!
-		Network:       network.TestNetworkPassphrase,
-	}
-
-	err = tx.Build()
+	tx, err := txnbuild.NewTransaction(
+		txnbuild.TransactionParams{
+			SourceAccount:        &sourceAccount,
+			IncrementSequenceNum: false,
+			Operations:           []txnbuild.Operation{&op},
+			BaseFee:              txnbuild.MinBaseFee,
+			Timebounds:           txnbuild.NewInfiniteTimeout(), // Use a real timeout in production!
+		},
+	)
```

Comment:
What benefits should we expect to see with putting the build logic in `NewTransaction` rather than keeping it in the `Build` function? It's not obvious to me what the benefit is. NewTransaction looks good, but I don't see it as being more ergonomic than what we have now.

As a bonus if we keep the `Build` function, and rename `TransactionParams` to `TransactionBuilder` the API for building will also look and behave mostly like the builder pattern which is quickly recognized by many folks and is the pattern we're using in other SDKs (e.g. Javascript SDK). Are we thinking of replacing the builder pattern in other SDKs with something like this?

```suggestion
	txb := txnbuild.TransactionBuilder{
		SourceAccount:        &sourceAccount,
		IncrementSequenceNum: false,
		Operations:           []txnbuild.Operation{&op},
		BaseFee:              txnbuild.MinBaseFee,
		Timebounds:           txnbuild.NewInfiniteTimeout(), // Use a real timeout in production!
	}
	tx, err := txb.Build()
```

_I thought I had raised this previously but I can't find my previous comment about this. Apologies if I'm re-raising something that's been discussed already._

Reply:
I prefer using `NewTransaction()` over having `TransactionBuilder` because if you don't care about keeping the builder around the code looks better with `NewTransaction()`
```golang
	tx, err := txnbuild.NewTransaction(
		txnbuild.TransactionParams{
			SourceAccount:        &sourceAccount,
			IncrementSequenceNum: false,
			Operations:           []txnbuild.Operation{&op},
			BaseFee:              txnbuild.MinBaseFee,
			Timebounds:           txnbuild.NewInfiniteTimeout(), // Use a real timeout in production!
		},
	)
```

vs

```golang
	tx, err := txnbuild.TransactionBuilder{
		SourceAccount:        &sourceAccount,
		IncrementSequenceNum: false,
		Operations:           []txnbuild.Operation{&op},
		BaseFee:              txnbuild.MinBaseFee,
		Timebounds:           txnbuild.NewInfiniteTimeout(), // Use a real timeout in production!
	}.build()
```

Reply:
this is how the builder pattern looks for the java sdk:

```java
        Transaction transaction = new Transaction.Builder(account, Network.TESTNET)
                .addOperation(new CreateAccountOperation.Builder(destination.getAccountId(), "2000").build())
                .addMemo(Memo.text("Hello world!"))
                .setTimeout(Transaction.Builder.TIMEOUT_INFINITE)
                .build();
```

Note how `TransactionBuilder` doesn't persist beyond this statement which creates a `Transaction`.

I do think it's nice for there to be consistency among all the horizon sdk implementations. However, we should be mindful that there can be some differences between the implementations because each language has its own set of styles and conventions.

In this particular case, I would argue that constructing a transaction using `txnbuild.NewTransaction()` in the Go SDK is a good translation of  `new Transaction.Builder()...build()` from the Java SDK.  It may appear that they're inconsistent on a cosmetic level but on a practical and semantic level they are effectively equivalent.

Reply:
I like the `NewTransaction` approach. I don't feel it's hard to understand what's happening with this naming.

Reply:
I would rather avoid dealing with a `Builder`, plus `NewTransaction()` fits a known pattern.

Reply:
👍